### PR TITLE
chore: Bump prometheus/common library

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
In later releases of prometheus/common FormatStrings got unexported to allow to check for more permutations with other features encoded in the contentType. 
https://github.com/prometheus/common/pull/576

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
None
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: pkg/metricshandler/metrics_handler.go [hunk 1]
```diff
@@ -189,8 +189,12 @@ func (m *MetricsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	contentType := expfmt.NegotiateIncludingOpenMetrics(r.Header)
 
 	// We do not support protobuf at the moment. Fall back to FmtText if the negotiated exposition format is not FmtOpenMetrics See: https://github.com/kubernetes/kube-state-metrics/issues/2022.
-	if contentType != expfmt.FmtOpenMetrics_1_0_0 && contentType != expfmt.FmtOpenMetrics_0_0_1 {
-		contentType = expfmt.FmtText
+
+	// prometheus/common/expfmt does not allow to check for OpenMetrics version, so check up for older version separately
+	openMetrics_0_0_1, _ := expfmt.NewOpenMetricsFormat(expfmt.OpenMetricsVersion_0_0_1)
```
Comment (mrueg) [inline pkg/metricshandler/metrics_handler.go:194]: This is not ideal, but FormatType only exposes TypeOpenMetrics which is fmtOpenMetrics_1_0_0

**Comment by mrueg** [comment]
> /lgtm
I noticed that you haven't been involved in this project, so this lgtm feels very drive-by-ish.
Would you like to become more involved with the project?

**Comment by knabben** [comment]
@mrueg It was for a specific case;  I apologize if you were expecting other assigned reviewers.

I'm not sure if I have the bandwidth to collaborate on the project actively.

**Comment by mrueg** [comment]
@knabben Yes, the expectation in general is that contributors would lgtm pull requests and not unrelated members of the kubernetes org (and if they do it, then they would usually provide some context why).

Was there any specific bug or issue that you saw that this PR fixes?

**Comment by shuguet** [comment]
Sorry folks, I reached out on the CNCF Ambassador channel to ask if there was any maintainer of this project present who could approved this, as our (internal) project includes some modules from this repo, and we were blocked (importing 2 modules of different versions, the usual).

I did not mean to cause any issue here 🙏 

Thank you for merging this though!

**Pull request merged**